pkg/kf/commands/install/gke.go

```diff
@@ -0,0 +1,908 @@
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/manifoldco/promptui"
+	"github.com/spf13/cobra"
+)
+
+const (
+	KnativeBuildYAML      = "https://github.com/knative/build/releases/download/v0.7.0/build.yaml"
+	KfNightlyBuildYAML    = "https://storage.googleapis.com/artifacts.kf-releases.appspot.com/nightly-builds/releases/release-latest.yaml"
+	GKEClusterVersion     = "1.13.7-gke.8"
+	MachineType           = "n1-standard-4"
+	ImageType             = "COS"
+	DiskType              = "pd-standard"
+	DiskSize              = "100"
+	NumNodes              = "3"
+	DefaultMaxPodsPerNode = "110"
+	Addons                = "HorizontalPodAutoscaling,HttpLoadBalancing,Istio,CloudRun"
+)
+
+var (
+	prefixColor = color.New(color.FgHiBlue, color.Bold)
+	labelColor  = color.New(color.FgHiYellow, color.Bold)
+)
+
+// NewGKECommand creates a command that can install kf to GKE+Cloud Run.
+// TODO: This installer is using gcloud and kubectl under the hood. We should
+// really replace gcloud with google.golang.org/api and kubectl with
+// k8s.io/client-go
+func NewGKECommand() *cobra.Command {
+	var verbose bool
+
+	cmd := &cobra.Command{
+		Use:   "gke [subcommand]",
+		Short: "Install kf to GKE with Cloud Run",
+		Long: `
+This interactive installer will walk you through a process to install GKE with
+Cloud Run and then kf. Its really just running scripts under the hood,
+therefore you have to have gcloud and kubectl installed and available in the
+path.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			ctx := setContextOutput(context.Background(), cmd.ErrOrStderr())
+			ctx = setLogPrefix(ctx, "Install GKE+CR+Kf")
+			ctx = setVerbosity(ctx, verbose)
+
+			// Select the desired project
+			projID, err := selectProject(ctx)
+			if err != nil {
+				return err
+			}
+
+			// Select the desired GKE cluster
+			clusterName, zone, err := selectCluster(ctx, projID)
+			if err != nil {
+				return err
+			}
+
+			// Target the Cluster
+			if err := targetCluster(
+				ctx,
+				clusterName,
+				zone,
+				projID,
+			); err != nil {
+				return err
+			}
+
+			// Install Knative Build
+			Logf(ctx, "install Knative Build")
+			if _, err := kubectl(
+				ctx,
+				"apply",
+				"--filename",
+				KnativeBuildYAML,
+			); err != nil {
+				return err
+			}
+
+			// Install Service Catalog
+			if err := installServiceCatalog(ctx); err != nil {
+				return err
+			}
+
+			// Install kf
+			Logf(ctx, "install kf")
+			if _, err := kubectl(
+				ctx,
+				"apply",
+				"--filename",
+				KfNightlyBuildYAML,
+			); err != nil {
+				return err
+			}
+
+			// Setup kf space
+			if err := setupSpace(ctx, projID); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().BoolVarP(
+		&verbose,
+		"verbose",
+		"v",
+		false,
+		"Display the gcloud and kubectl commands",
+	)
+
+	return cmd
```

Comment:
Something for thought: it would be really great if we could accumulate destroy commands for anything we created and print them out as part of a post-run so a user could tear down in case something went wrong

Reply:
https://github.com/google/kf/issues/455